src/mrpro/data/SpatialDimension.py

```diff
@@ -3,16 +3,38 @@
 from __future__ import annotations
 
 from collections.abc import Callable
+from copy import deepcopy
 from dataclasses import dataclass
-from typing import Generic, Protocol, TypeVar
+from typing import Any, Generic, Protocol, TypeVar, get_args, overload
 
 import numpy as np
 import torch
 from numpy.typing import ArrayLike
 
+import mrpro.utils.typing as type_utils
 from mrpro.data.MoveDataMixin import MoveDataMixin
 
-T = TypeVar('T', int, float, torch.Tensor)
+# Change here to add more types
+VectorTypes = torch.Tensor
+ScalarTypes = int | float
+T = TypeVar('T', torch.Tensor, int, float)
+T_co = TypeVar('T_co', torch.Tensor, int, float, covariant=True)
+T_co_float = TypeVar('T_co_float', float, torch.Tensor, covariant=True)
+T_co_vector = torch.Tensor
+T_co_scalar = TypeVar('T_co_scalar', int, float, covariant=True)
+
+
+def _as_vectortype(x: ArrayLike) -> VectorTypes:
+    """Convert ArrayLike to VectorType."""
+    if isinstance(x, VectorTypes) and type(x) in get_args(VectorTypes):
```

Comment:
is `type(x) in get_args(VectorTypes)` not enough here?

Reply:
At runtime yes. But mypy curently only understands the former.